src/lib/mina_base/party.ml

```diff
@@ -851,343 +992,53 @@ module Body = struct
-
-  let%test_unit "json roundtrip accept" =
-    let account_precondition : t = Accept in
-    let module Fd = Fields_derivers_zkapps.Derivers in
-    let full = deriver (Fd.o ()) in
-    [%test_eq: t] account_precondition
-      (account_precondition |> Fd.to_json full |> Fd.of_json full)
-
-  let%test_unit "json roundtrip nonce" =
-    let account_precondition : t = Nonce (Account_nonce.of_int 928472) in
-    let module Fd = Fields_derivers_zkapps.Derivers in
-    let full = deriver (Fd.o ()) in
-    [%test_eq: t] account_precondition
-      (account_precondition |> Fd.to_json full |> Fd.of_json full)
-
-  let%test_unit "json roundtrip full" =
-    let n = Account_nonce.of_int 4513 in
-    let account_precondition : t =
-      Full
-        { Zkapp_precondition.Account.accept with
-          nonce = Check { lower = n; upper = n }
-        ; public_key = Check Public_key.Compressed.empty
-        }
-    in
-    let module Fd = Fields_derivers_zkapps.Derivers in
-    let full = deriver (Fd.o ()) in
-    [%test_eq: t] account_precondition
-      (account_precondition |> Fd.to_json full |> Fd.of_json full)
-
-  let%test_unit "to_json" =
-    let account_precondition : t = Nonce (Account_nonce.of_int 34928) in
-    let module Fd = Fields_derivers_zkapps.Derivers in
-    let full = deriver (Fd.o ()) in
-    [%test_eq: string]
-      (account_precondition |> Fd.to_json full |> Yojson.Safe.to_string)
-      ( {json|{
-          balance: null,
-          nonce: {lower: "34928", upper: "34928"},
-          receiptChainHash: null, publicKey: null, delegate: null,
-          state: [null,null,null,null,null,null,null,null],
-          sequenceState: null, provedState: null
-        }|json}
-      |> Yojson.Safe.from_string |> Yojson.Safe.to_string )
-
-  let digest (t : t) =
-    let digest x =
-      Random_oracle.(
-        hash ~init:Hash_prefix_states.party_account_precondition (pack_input x))
-    in
-    to_full t |> Zkapp_precondition.Account.to_input |> digest
-
-  module Checked = struct
-    type t = Zkapp_precondition.Account.Checked.t
-
-    let digest (t : t) =
-      let digest x =
-        Random_oracle.Checked.(
-          hash ~init:Hash_prefix_states.party_account_precondition
-            (pack_input x))
-      in
-      Zkapp_precondition.Account.Checked.to_input t |> digest
-  end
-
-  let typ () : (Zkapp_precondition.Account.Checked.t, t) Typ.t =
-    Typ.transport (Zkapp_precondition.Account.typ ()) ~there:to_full
-      ~back:(fun s -> Full s)
+module Poly (Body : Type) (Auth : Type) = struct
+  type t = { body : Body.t; authorization : Auth.t }
 end
 
-module Preconditioned = struct
-  module Poly = struct
-    [%%versioned
-    module Stable = struct
-      module V1 = struct
-        type ('body, 'account_precondition) t =
-          { body : 'body; account_precondition : 'account_precondition }
-        [@@deriving annot, hlist, sexp, equal, yojson, hash, compare, fields]
-      end
-    end]
-  end
-
+module T = struct
```

Comment:
Curious why you had to add a wrapping `T`module here that's included later!

Reply:
To specify the return type in `Fee_payer.to_party`